bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexServerHandler.java

```diff
@@ -0,0 +1,307 @@
+                        if (tmpMeta.getType().equals("episode")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getGrandparentThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " : " + tmpMeta.getTitle());
+                        } else if (tmpMeta.getType().equals("track")) {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                            tmpMeta.setTitle(tmpMeta.getGrandparentTitle() + " - " + tmpMeta.getParentTitle() + " - "
+                                    + tmpMeta.getTitle());
+                        } else {
+                            tmpMeta.setThumb(plexAPIConnector.getURL(tmpMeta.getThumb()));
+                        }
+                        playerHandlers.get(tmpMeta.getPlayer().getMachineIdentifier()).refreshSessionData(tmpMeta);
+                        playerActiveCount++;
+                    }
+                }
+            }
+        }
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNT),
+                new StringType(String.valueOf(playerCount)));
+        updateState(new ChannelUID(getThing().getUID(), PlexBindingConstants.CHANNEL_SERVER_COUNTACTIVE),
+                new StringType(String.valueOf(playerActiveCount)));
+    }
+
+    /**
+     * Refresh all the configured players
+     */
+    private void refreshAllPlayers() {
+        Iterator<PlexPlayerHandler> valueIterator = playerHandlers.values().iterator();
+        while (valueIterator.hasNext()) {
+            valueIterator.next().updateChannels();
+        }
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            int pollingInterval = ((BigDecimal) getConfig().get(CONFIG_REFRESH_RATE)).intValue();
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1, pollingInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * The refresh job, pulls the session data and then calls refreshAllPlayers which will have them send
+     * out their current status.
+     */
+    private Runnable pollingRunnable = () -> {
+        try {
+            MediaContainer plexSessionData = plexAPIConnector.getSessionData();
+            if (plexSessionData != null) {
+                refreshStates(plexSessionData);
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Plex is not returning valid session data");
+            }
+            refreshAllPlayers();
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, System.out
```

Comment:
I saw this while compiling a jar of this to play with.  Is System.out.printf() appropriate here?  I would think it should be new String() or something similar to that.

Reply:
Thanks for the feedback, I've updated the code